File: app/src/main/java/com/xmx/tango/module/tango/AddTangoActivity.java
package com.xmx.tango.module.tango;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.View;
import android.widget.EditText;

import com.xmx.tango.R;
import com.xmx.tango.base.activity.BaseTempActivity;

import org.greenrobot.eventbus.EventBus;
import org.xutils.view.annotation.ContentView;
import org.xutils.view.annotation.Event;
import org.xutils.view.annotation.ViewInject;

import java.util.Date;

/**
 * A simple {@link Fragment} subclass.
 */
@ContentView(R.layout.activity_add_tango)
public class AddTangoActivity extends BaseTempActivity {

    @ViewInject(R.id.edit_writing)
    EditText writingView;

    @ViewInject(R.id.edit_pronunciation)
    EditText pronunciationView;

    @ViewInject(R.id.edit_meaning)
    EditText meaningView;

    @ViewInject(R.id.edit_tone)
    EditText toneView;

    @ViewInject(R.id.edit_part_of_speech)
    EditText partOfSpeechView;

    @ViewInject(R.id.edit_type)
    EditText typeView;

    @Event(value = R.id.btn_add_tango)
    private void onAddTangoClick(View view) {
        Tango entity = new Tango();

        String writing = writingView.getText().toString().trim();
        entity.writing = writing;

        String pronunciation = pronunciationView.getText().toString().trim();
        entity.pronunciation = pronunciation;

        String meaning = meaningView.getText().toString().trim();
        entity.meaning = meaning;

        String toneStr = toneView.getText().toString().trim();
        int tone = -1;
        if (!toneStr.equals("")) {
            try {
                tone = Integer.parseInt(toneStr);
            } catch (Exception e) {
                showToast("音调格式不合法");
                return;
            }
        }
        entity.tone = tone;

        String partOfSpeech = partOfSpeechView.getText().toString().trim();
        entity.partOfSpeech = partOfSpeech;

        String type = typeView.getText().toString().trim();
        entity.type = type;

        if (writing.equals("") && pronunciation.equals("")) {
            showToast(R.string.add_error);
            return;
        }

        entity.addTime = new Date();
        TangoEntityManager.getInstance().insertData(entity);

        writingView.setText("");
        pronunciationView.setText("");
        meaningView.setText("");
        toneView.setText("");
        partOfSpeechView.setText("");
        typeView.setText("");

        showToast(R.string.add_success);

        EventBus.getDefault().post(new TangoListChangeEvent());
    }

    @Override
    protected void initView(Bundle savedInstanceState) {

    }

    @Override
    protected void setListener() {

    }

    @Override
    protected void processLogic(Bundle savedInstanceState) {

    }
}

File: app/src/main/java/com/xmx/tango/module/crud/AddTangoActivity.kt
package com.xmx.tango.module.crud

import android.os.Bundle

import com.xmx.tango.R
import com.xmx.tango.base.activity.BaseTempActivity
import com.xmx.tango.module.tango.Tango
import com.xmx.tango.module.tango.TangoEntityManager
import kotlinx.android.synthetic.main.activity_add_tango.*

import org.greenrobot.eventbus.EventBus

import java.util.Date

/**
 * Created by The_onE on 2017/10/8.
 * 添加单语Activity
 */
class AddTangoActivity : BaseTempActivity() {

    override fun initView(savedInstanceState: Bundle?) {
        setContentView(R.layout.activity_add_tango)
    }

    override fun setListener() {
        btnAddTango.setOnClickListener {
            val entity = Tango()
            // 写法
            val writing = writingView.text.toString().trim()
            entity.writing = writing
            // 发音
            val pronunciation = pronunciationView.text.toString().trim()
            entity.pronunciation = pronunciation
            // 解释
            entity.meaning = meaningView.text.toString().trim()
            // 音调
            val toneStr = toneView.text.toString().trim()
            var tone = -1
            if (toneStr.isNotBlank()) {
                try {
                    tone = Integer.parseInt(toneStr)
                } catch (e: Exception) {
                    showToast("音调格式不合法")
                    return@setOnClickListener
                }
            }
            entity.tone = tone
            // 词性
            val partOfSpeech = partOfSpeechView.text.toString().trim()
            entity.partOfSpeech = partOfSpeech
            // 类型
            val type = typeView.text.toString().trim()
            entity.type = type
            // 写法和发音不能同时为空
            if (writing == "" && pronunciation == "") {
                showToast(R.string.add_error)
                return@setOnClickListener
            }
            // 添加单语
            entity.addTime = Date()
            TangoEntityManager.insertData(entity)
            // 清空编辑框
            writingView.setText("")
            pronunciationView.setText("")
            meaningView.setText("")
            toneView.setText("")
            partOfSpeechView.setText("")
            typeView.setText("")

            showToast(R.string.add_success)

            EventBus.getDefault().post(TangoListChangeEvent())
        }
    }

    override fun processLogic(savedInstanceState: Bundle?) {

    }
}
